cmd/notation/plugin/install.go

```diff
@@ -330,6 +331,16 @@ func installPluginWithOptions(ctx context.Context, opts plugin.CLIInstallOptions
 		if errors.As(err, &errPluginDowngrade) {
 			return fmt.Errorf("%w.\nIt is not recommended to install an older version. To force the installation, use the \"--force\" option", errPluginDowngrade)
 		}
+
+		var errExeFile *plugin.PluginExecutableFileError
+		if errors.As(err, &errExeFile) {
+			return fmt.Errorf("%w.\nPlease ensure the plugin is executable on %s/%s.", err, runtime.GOOS, runtime.GOARCH)
+		}
+
+		var errMalformedPlugin *plugin.PluginMalformedError
+		if errors.As(err, &errMalformedPlugin) {
+			return fmt.Errorf("%w.\nPlease ensure to install the plugin for %s/%s. Contact the plugin publisher if the plugin executable file is malformed.", errMalformedPlugin, runtime.GOOS, runtime.GOARCH)
```

Comment:
```suggestion
			return fmt.Errorf("%w.\n The %s plugin is malformed. Please contact the plugin publisher for assistance.", errMalformedPlugin, runtime.GOOS, runtime.GOARCH)
```

Reply:
@priteshbandi There are two circumstances that can trigger `PluginMalformedError`:
1. `User error`: the user is trying to install a plugin for another arch. For example, installing `arm64` plugin file on `amd64` arch. In this case, the plugin itself is not malformed, the user needs to install the correct file.
2. `Plugin publisher error`: the plugin publisher published a plugin that's not following Notary project's spec. For example, notation cannot unmarshal the plugin response. In this case, the plugin itself is malformed, and the only actionable suggestion from Notation CLI is to contact the plugin publisher.

Given above, my suggestion is:
```suggestion
			return fmt.Errorf("%w.\nPlease ensure that the plugin executable file is intact and compatible with %s/%s. Contact the plugin publisher for further assistance.", errMalformedPlugin, runtime.GOOS, runtime.GOARCH)
```

Reply:
```
./bin/notation plugin install --url https://github.com/Azure/notation-azure-kv/releases/download/v1.0.1/notation-azure-kv_1.0.1_linux_arm64.tar.gz --sha256sum d6b4be3cef6a5cb154ddd73f7bdc83f5d7e913e3b72eb88f7a5f82872ca63a5a -d
Downloading plugin from https://github.com/Azure/notation-azure-kv/releases/download/v1.0.1/notation-azure-kv_1.0.1_linux_arm64.tar.gz
Download completed
DEBU[2024-01-19T09:40:05+08:00] Extracting file notation-azure-kv...         
DEBU[2024-01-19T09:40:05+08:00] Extracting file LICENSE...                   
DEBU[2024-01-19T09:40:05+08:00] Installing plugin from path /tmp/notation-plugin1234095091 
DEBU[2024-01-19T09:40:05+08:00] Plugin get-plugin-metadata request: {}       
ERRO[2024-01-19T09:40:05+08:00] plugin get-plugin-metadata execution status: exit status 255 
Error: plugin installation failed: failed to execute the get-plugin-metadata command for plugin azure-kv: qemu-aarch64: Could not open '/lib/ld-linux-aarch64.so.1': No such file or directory.
Please ensure to install the plugin for linux/amd64. Contact the plugin publisher if the plugin executable file is malformed.
```

I encountered an error: 'qemu-aarch64: Could not open '/lib/ld-linux-aarch64.so.1': No such file or directory', which was classified as `PluginMalformedError`. This classification occurred because the error was returned from `stderr`. The error arose due to the user enabling the ARM qemu emulator without installing the required library on a x64 machine. In other words, a PluginMalformedError may also be caused by an architecture mismatch issue. Therefore, it is useful to include the `platform/Architecture` in the message to assist users in debugging the issue.